pyccel/parser/semantic.py

```diff
@@ -443,15 +443,20 @@ def get_function(self, name):
 
 
     def get_import(self, name):
-        """."""
+        """
+        Search for an import with the given name in the current namespace.
+        Return None if not found.
+        """
 
         imp = None
-
         container = self.namespace
         while container:
 
-            if name in container.imports['imports']:
-                imp =  container.imports['imports'][name]
+            for key,_ in container.imports['imports'].items():
+                if key == name:
```

Comment:
This can be simplified to
```
if name in container.imports['imports']:
```

Reply:
1- having import like 
```PYTHON
import numpy as np
import math as np
```
and a call
```
x = np.sqrt()
```
both of those import satisfy the condition 
``` 
if name in container.imports['imports']:
#or
key == name    #AsName(numpy, np ) == np and AsName(math, np) == np
```
so we need to take the last import created that's the reason I'm looping
is there another way to collect the last element in dict that satisfy a given condition?

Reply:
A dictionary can't have multiple elements with the same key. If the key is the same then the first import will no longer exist in the dictionary after the import of the second element so you will automatically get the last object added

Reply:
I said nothing about the same key.  here key can be an AsName "numpy as np" and "math as np" both have np as target so we can track dotted elements np.xxxx

Reply:
By "same key" I mean that the dictionary considers it to be the same. The details of how this is done are discussed [here](https://stackoverflow.com/questions/32279666/objects-are-not-considered-the-same-in-dictionary-keys-but-eq-is-implement). This means that if the class has no `__eq__` method then it will work as you describe (`numpy as np` and `math as np` will be considered to be different keys), but if it has an `__eq__` method (as for `AsName`) then it depends on the result of the `__hash__` function. For [`AsName`](https://github.com/pyccel/pyccel/blob/master/pyccel/ast/core.py#L600) this means that the key is essentially the target so `numpy as np` and `math as np` will be considered to be the same key. You can verify this by printing the dictionary